source/secure_tunneling.c

```diff
@@ -600,8 +638,11 @@ static void s_aws_secure_tunnel_on_stream_start_received(
      */
     s_set_absent_connection_id_to_one(message_view, &connection_id);
 
-    int result =
-        s_aws_secure_tunnel_set_stream(secure_tunnel, message_view->service_id, message_view->stream_id, connection_id);
+    int result = AWS_OP_SUCCESS;
```

Comment:
While both integers, you shouldn't combine AWS_OP_SUCCESS/AWS_OP_ERR with error codes.

Presumable this should be:

```
int error_code = AWS_ERROR_SUCCESS;
if (...) {
   error_code = aws_last_error();
}

```

Reply:
Gotcha. I don't know exactly why but `AWS_ERROR_SUCCESS` makes me feel bad >,<